src/orange/orangeinp/UnitProto.cc

```diff
@@ -182,7 +185,9 @@ void UnitProto::build(ProtoBuilder& input) const
         VolumeInput vi;
 
         // Construct logic and faces with remapped surfaces
-        auto&& [faces, logic] = build_logic(node_id);
+        auto&& [faces, logic]
+            = build_logic.operator()<detail::PostfixLogicBuilderPolicy>(
```

Comment:
If we move the instantiation out, of the `LogicBuilder` class, we'd still have to be templated on the `Policy` type so the call would still be very verbose. Given that the builder holds all the objects necessary to construct the policy rather than asking the user to create a new policy object every time we call the builder.

Reply:
Wouldn't it look like `build_logic(PostfixLogicBuilderPolicy{}, node_id)`? Maybe `LogicBuilder` should actually be a `class PostfixLogicBuilder : public LogicBuilder<PostfixLogicBuilder>`?

Reply:
I went with the suggestion of having the policy take the construction argument, the `LogicBuilder` could be a free function now since it no longer has a state and a single call operator.